### PR TITLE
Introduced the Collection data type object

### PR DESCRIPTION
**Pull request by jcomack**
This will allows for Assessors to have a Collection of assessments that need to be processed.

### CODE REVIEW

File: js/datatypes/Collection.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ */
+Collection.prototype.removeItem = function( index ) {
+	if ( indexExists( this._items, index ) === false ) {
+		return false;
+	}
+
+	this._items.splice( index, 1 );
+	this._length = calculateLength( this._items );
+
+	return true;
```
Comment (terw-dan) [inline js/datatypes/Collection.js]: If you remove the item where the current iterator is pointed to, the iterator will point the a next item. Something we should check for.

**Comment by terw-dan** [comment]
CR 1 Done

**Comment by jcomack** [comment]
After discussing with @atimmer, we came to the conclusion that it might be easier to use a native `Array` until `Set` methods are available in Babel or something similar to Babel.

**Comment by atimmer** [comment]
We can close this, the code will be preserved.

**Pull request closed without merging**